tinygrad/codegen/linearizer.py

```diff
@@ -378,15 +366,16 @@ def end_loop(xx:List[Variable]):
         # end the late reduce loop
         end_loop(end_local_idxs)
         self.load_cache.clear()
+        acc_idxs = global_idxs+local_idxs+fake_reduce_idxs+upcast_idxs
 
     # load latebufs
-    loaded_buffers.update({b:self.global_load(i, global_idxs+local_idxs+fake_reduce_idxs+upcast_idxs) for i,b in enumerate(self.bufs) if b not in self.earlybufs and i != 0 and b.__class__ is not LocalBuffer})
+    loaded_buffers.update({b:self.global_load(i, acc_idxs) for i,b in enumerate(self.bufs) if b not in self.earlybufs and i != 0 and b.__class__ is not LocalBuffer})
```

Comment:
```suggestion
    loaded_buffers.update({b:self.global_load(i, acc_idxs) for i,b in enumerate(self.bufs[1:], start=1) if b not in self.earlybufs and b.__class__ is not LocalBuffer})
```

Reply:
This was just a hack to make it work - I'll remove this when there's a good way to linked main buffer with various local ones.